### PR TITLE
Feature/post page

### PR DESCRIPTION
**Pull request by EmilieSchott**
Hello,

I add the post page feature in my blog project. Could you tell me what do you think ? 

Thank you very much. :)

### CODE REVIEW

File: controller/PublicController.php [hunk 1]
```diff
@@ -68,24 +68,66 @@ public function contactMe() {
...
+            
+            if (array_key_exists('entry', $datas) && ($datas['entry']<0 || $datas['entry']>1)) {
+                unset($datas['entry']);
+            }
+
+            echo $twig->render('postView.html.twig', $datas);
+        } else {
+            header('Location: index.php?action=blog&page=1');
+        }
+    }
```
Comment (Tiriel) [inline controller/PublicController.php]: You should split all this public function into smaller private functions, called by the main public function. Something along the lines of this:
```php
    public function post(PostManager $postManager, CommentManager $commentManager, Twig $twig, array $datas) {
        if ($datas['id']>0) {
        	$this->getPostAndComments($datas);
        }
        // other things
    }
    
    private function getPostAndComment(array &$datas): array
    {
        $post = $postManager->getPost($datas['id']);
        $datas['post'] = $post[0];
        if (is_null($datas['post'])) {
            header('Location: index.php?action=blog&page=1');
        }

        $comments = $commentManager->getComments($datas['id']);
        if (!is_null($comments['pagesNbr'])) {
            $datas['pagesNbr'] = $comments['pagesNbr'];
            if (!array_key_exists('page', $datas) || $datas['page']<=0 || $datas['page']>$datas['pagesNbr']) {
                $datas['page']=1;
            }
            $datas['comments'] = $commentManager->accessPage($comments['datasPages'], $datas['page']);
		}
	}
```

File: controller/PublicController.php [hunk 2]
```diff
@@ -68,24 +68,66 @@ public function contactMe() {
...
+                $datas['pagesNbr'] = $comments['pagesNbr'];
+                if (!array_key_exists('page', $datas) || $datas['page']<=0 || $datas['page']>$datas['pagesNbr']) {
+                    $datas['page']=1;
+                }
+                $datas['comments'] = $commentManager->accessPage($comments['datasPages'], $datas['page']);
+            } else {
+                if (array_key_exists('page', $datas)) {
+                    unset($datas['page']);
+                }
+            }
```
Comment (Tiriel) [inline controller/PublicController.php]: Can be simplified with one `else if (...)`

File: index.php [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 use EmilieSchott\BlogPHP\Controller\PublicController;
+use EmilieSchott\BlogPHP\Model\PostManager;
+use EmilieSchott\BlogPHP\Model\CommentManager;
 
 require_once __DIR__ . '/vendor/autoload.php';
 
 $loader = new \Twig\Loader\FilesystemLoader(__DIR__ . '/view');
 $twig = new \Twig\Environment($loader, [
-    'cache' => false // "False" to replace by " __DIR__ . '/tmp' " to enable cache.
+    'cache' => false // TO DO : "False" to replace by " __DIR__ . '/tmp' " to enable cache.
```
Comment (Tiriel) [inline index.php]: Not needed, TODO means this has to be done absolutely at one point. Cache is a good measure, but you're under no obligation to use it.

File: model/Manager.php [hunk 4]
```diff
@@ -2,33 +2,15 @@
 
 namespace EmilieSchott\BlogPHP\Model;
 
+use EmilieSchott\BlogPHP\Paginator\Paginator;
 use EmilieSchott\BlogPHP\Model\PDOFactory;
 
-abstract class Manager {
+abstract class Manager extends Paginator {
```
Comment (Tiriel) [inline model/Manager.php]: Why extend if you create a paginator inside the construct?

**Review by Tiriel — COMMENTED** [review]
You should try to configure properly cs-fixer, you have some consistency problems with PSR 2/PSR 12 and coding style (spaces around equal signs, brackets on new lines, etc)

File: controller/PublicController.php (hunk 2 above)
Comment (EmilieSchott) [reply controller/PublicController.php]: I really don't understand why I wrote that... Sorry ! I'm fixing it right now.

File: index.php (hunk 3 above)
Comment (EmilieSchott) [reply index.php]: Okay, I understand the nuance. Thank you.

File: model/Manager.php (hunk 4 above)
Comment (EmilieSchott) [reply model/Manager.php]: Hm... I assume this is an oversight of one of my previous attempts to organize my code. I'm removing it.

File: controller/PublicController.php (hunk 1 above)
Comment (EmilieSchott) [reply controller/PublicController.php]: Ok, I' seeing your point

**Comment by EmilieSchott** [comment]
> 
> 
> You should try to configure properly cs-fixer, you have some consistency problems with PSR 2/PSR 12 and coding style (spaces around equal signs, brackets on new lines, etc)

You're right ! php-cs-fixer was installed but not configured correctly in VS but it fixed now. It's not a bad thing since you could point to me what I'm doing wrong in regard to PSR2/PSR12. ;)

**Review by Tiriel — APPROVED** [review]

**Pull request merged**